cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1580,6 +1834,80 @@ data TxBodyContent build era =
      }
      deriving (Eq, Show)
 
+instance
+  ( Default (BuildTxWith build (Maybe ProtocolParameters))
+  , IsCardanoEra era
+  ) => Default (TxBodyContent build era) where
+  def = TxBodyContent
```

Comment:
I read an article in the past why `Default` (or rather `def`) is actually not so nice to use after all.. can't find it back, like a commenter on this thread: https://discourse.haskell.org/t/do-people-actually-use-data-default-data-default-default-type-class-in-production/3951

Please consider especially the point that you are writing a library here which already has like 400 dependencies. Adding one more (`data-default-class`), especially downstream to users is not helping .. at least re-export `def` if you decide to go the `Default` way.

To me, as a user, a single `defaultTxBodyContent` would be perfectly fine. Maybe even call it `emptyTxBodyContent` if you want. We don't need any of the other `def` values as the `TxMintNone` etc are good enough and don't introduce "blindness" as `Nothing`s or `def`s would do.

Reply:
> Default (or rather def) is actually not so nice to use after all

This is not a universally accepted view. It is a subjective opinion of some developers. However, there is nothing wrong with providing specialized functions, like `emptyTxBodyContent`, along with `Default` instances. This would give the user a choice of preference, precisely because not everyone like `Default`.

> Please consider especially the point that you are writing a library here which already has like 400 dependencies.

`data-default-class` is already a transitive dependency, so we are not adding any new dependencies in this PR

Reply:
> "It's already a transitive dependency" is exactly how you get into 400 dependencies territory.

Sorry, I don't get it. How do you get to `n` dependencies by not adding a new dependency?

Reply:
> less code to maintain

This is my main contention. I also think defining a `defaultTxBodyContent` value is sufficient. I'd rather not introduce more classes/types if it's not absolutely necessary.